tests/test_generate.py

```diff
@@ -0,0 +1,138 @@
+
+new_notebook = {"cells": [], "metadata": {}, "nbformat": 4, "nbformat_minor": 4}
+new_notebook = json.dumps(new_notebook, indent=4)
+
+
+notebook_with_content = {
+    "cells": [
+        {
+            "cell_type": "code",
+            "execution_count": 1,
+            "metadata": {},
+            "outputs": [
+                {
+                    "name": "stdout",
+                    "output_type": "stream",
+                    "text": ["2.302585092994046\n"],
+                }
+            ],
+            "source": ["import math \n", "print(math.log(10))"],
+        }
+    ],
+    "metadata": {
+        "kernelspec": {
+            "display_name": "Python 3",
+            "language": "python",
+            "name": "python3",
+        },
+        "language_info": {
+            "codemirror_mode": {"name": "ipython", "version": 3},
+            "file_extension": ".py",
+            "mimetype": "text/x-python",
+            "name": "python",
+            "nbconvert_exporter": "python",
+            "pygments_lexer": "ipython3",
+            "version": "3.8.5",
+        },
+    },
+    "nbformat": 4,
+    "nbformat_minor": 4,
+}
+notebook_with_content = json.dumps(notebook_with_content, indent=4)
+
+
+cell_front_matter = {
+    "cell_type": "markdown",
+    "metadata": {},
+    "source": [
+        "<!-- ---\n",
+        "title: <required-title>\n",
+        f"date: {today}\n",
+        "downloads: true\n",
+        "weight: <required-weight> See: https://github.com/pytorch-ignite/examples/issues/30\n",
+        "summary: <use either this or the `<!--more-->` tag below to provide summary for this notebook, "
+        "and delete the other>\n"
+        "tags:\n",
+        "  - <required-tag>\n",
+        "--- -->\n",
+        "\n",
+        "# title-placeholder\n",
+        "\n",
+        "<If you are not using the `summary` variable above, use this space to "
+        "provide a summary for this notebook.>\n",
+        "<Otherwise, delete the `<!--more-->` below.>",
+        "\n",
+        "<!--more-->",
+    ],
+}
+
+
+cwd = os.getcwd()
+
+
+@pytest.mark.parametrize("name", ["dummy_notebook", "dummy_notebook.ipynb"])
+def test_new_notebook_creation(name):
+    output = subprocess.check_output(["python", "../generate.py", name]).decode("utf-8")
+
+    if not name.endswith(".ipynb"):
+        name = name + ".ipynb"
+
+    assert output == f"Generated {os.path.join(cwd, name)}\n"
+    subprocess.call(["rm", "-rf", name])
+
+
+def test_existing_blank_notebook():
+    name = "dummy_notebook_empty.ipynb"
+    with open(name, "w") as f:
+        f.write(new_notebook)
+
+    output = subprocess.check_output(["python", "../generate.py", name]).decode("utf-8")
+    assert output == f"Added frontmatter to {os.path.join(cwd, name)}\n"
+    subprocess.call(["rm", "-rf", name])
+
+
+def test_existing_non_empty_notebook():
+    name = "dummy_notebook_with_content.ipynb"
+    with open(name, "w") as f:
+        f.write(notebook_with_content)
```

Comment:
I was mentioning existing notebook like https://github.com/pytorch-ignite/examples/blob/main/tutorials/beginner/01-getting-started.ipynb
Here you create a notebook from code, by the way, you can use pytest fixtures like `tmpdir` (https://docs.pytest.org/en/6.2.x/builtin.html?highlight=tempdir)

Can we use them for that ?

Reply:
Didn't know about pytest's ``tmpdir``, will check it out!
So I should make a copy of an existing notebook in the pytest ``tmpdir``, and use that for testing right?
A copy is necessary as generate makes changes to the script itself (doesn't create new one as output), but don't want to change existing notebooks.

Reply:
Yes, copying is ok with `shutil.copyfile`, https://docs.python.org/3/library/shutil.html#shutil.copyfile

Reply:
I have replaced the existing notebook creation with this. I still have to create empty notebook for a few tests, should I also replace those?